lib/sycamore/sycamore/connectors/elasticsearch/elasticsearch_writer.py

```diff
@@ -0,0 +1,140 @@
+from dataclasses import dataclass, field
+from typing import Any, Optional
+from sycamore.connectors.common import drop_types, flatten_data
+from typing_extensions import TypeGuard
+
+from sycamore.data.document import Document
+from sycamore.connectors.base import BaseDBWriter
+
+from elasticsearch import Elasticsearch, ApiError
+from elasticsearch.helpers import parallel_bulk
+
+
+@dataclass
+class ElasticClientParams(BaseDBWriter.ClientParams):
+    url: str
+    es_client_args: dict = field(default_factory=lambda: {})
+
+
+@dataclass
+class ElasticTargetParams(BaseDBWriter.TargetParams):
+    index_name: str
+    mappings: dict[str, Any] = field(
+        default_factory=lambda: {
```

Comment:
does elastic have index settings? would we want to expose them?

Reply:
Yep, have included a way to expose them